homeassistant/components/binary_sensor/tod.py

```diff
@@ -0,0 +1,234 @@
+"""
+Support for showing the binary sensors represending current time of the day.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.tod/
+
+"""
+from datetime import datetime, timedelta
+import logging
+import pytz
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_NAME, CONF_AFTER, CONF_BEFORE,
+    SUN_EVENT_SUNRISE, SUN_EVENT_SUNSET)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.helpers.sun import (
+    get_astral_event_date, get_astral_event_next)
+from homeassistant.util import dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AFTER_OFFSET = 'after_offset'
+CONF_BEFORE_OFFSET = 'before_offset'
+
+ATTR_NEXT_UPDATE = 'next_update'
+
+
+SENSOR_SCHEMA = vol.Schema({
+
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_AFTER): vol.Any(cv.time, vol.All(
+        vol.Lower, cv.sun_event)),
+    vol.Optional(CONF_AFTER_OFFSET, default=timedelta(0)): cv.time_period,
+    vol.Required(CONF_BEFORE): vol.Any(cv.time, vol.All(
+        vol.Lower, cv.sun_event)),
+    vol.Optional(CONF_BEFORE_OFFSET, default=timedelta(0)): cv.time_period,
+    vol.Required(CONF_NAME): cv.string
```

Comment:
Please keep it ordered.

Reply:
@fabaff Ordered by what? Don't understand the reason. Should I put CONF_NAME first? What is the implication?

Reply:
I moved the CONF_NAME to the beginning. CONF_AFTER logically is first. Because the sensor is "ON" AFTER certain point in the time and BEFORE another point in the time. Offsets are logically  belonging to corresponding AFTER and BEFORE. Maybe there is another criteria I don't understand.